### PR TITLE
mcontrolcenter: init at 0.4.1

### PR DESCRIPTION
**Pull request by nadimkobeissi**
MControlCenter provides a relatively full-featured control panel for MSI laptops, written in Qt. It allows users to control MSI laptop power plans, fan speed, battery charging limits, keyboard backlight settings, USB power sharing and more.

The application comes with strong desktop integration: shortcuts, taskbar item, etc. etc. -- screenshots here:
https://github.com/dmitry-s93/MControlCenter/blob/main/README.md

This package has been previous requested by the NixOS community: https://github.com/NixOS/nixpkgs/issues/244881

The package comes with a patch that corrects MControlCenter's path to the `modprobe` binary, allowing it to automatically load the `ec_sys` kernel module (included with NixOS by default) with the right settings when launched.

This NixOS package has been fully tested on a NixOS Plasma5 desktop.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/mcontrolcenter/derivation.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    mkdir -p $out/share/dbus-1/system-services
+    echo "[D-BUS Service]" >> $out/share/dbus-1/system-services/mcontrolcenter.helper.service
+    echo "Name=mcontrolcenter.helper" >> $out/share/dbus-1/system-services/mcontrolcenter.helper.service
+    echo "Exec=$out/libexec/mcontrolcenter-helper" >> $out/share/dbus-1/system-services/mcontrolcenter.helper.service
+    echo "User=root" >> $out/share/dbus-1/system-services/mcontrolcenter.helper.service
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/dmitry-s93/MControlCenter";
+    description = "An application that allows you to change the settings of MSI laptops running Linux.";
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/mcontrolcenter/derivation.nix:57]:
```suggestion
    description = "Tool to change the settings of MSI laptops running Linux";
```

File: pkgs/os-specific/linux/mcontrolcenter/derivation.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    licencse = licenses.gpl3;
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/mcontrolcenter/derivation.nix:58]:
```suggestion
    licencse = licenses.gpl3Plus;
```
https://github.com/dmitry-s93/MControlCenter/blob/main/src/main.cpp#L8

File: pkgs/os-specific/linux/mcontrolcenter/derivation.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  };
+
+  patches = [ ./modprobe.patch ];
+
+  cmakeFlags = [
+    "-DENABLE_TESTING=OFF"
+    "-DENABLE_INSTALL=ON"
+  ];
+
+  installPhase = ''
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/mcontrolcenter/derivation.nix:43]:
```suggestion
  installPhase = ''
    runHook preInstall
```

File: pkgs/os-specific/linux/mcontrolcenter/derivation.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    install -Dm755 mcontrolcenter $out/bin/mcontrolcenter
+    install -Dm755 helper/mcontrolcenter-helper $out/libexec/mcontrolcenter-helper
+    install -Dm644  $src/resources/mcontrolcenter.svg $out/share/icons/hicolor/32x32/apps/mcontrolcenter.svg
+    install -Dm644 $src/src/helper/mcontrolcenter-helper.conf $out/share/dbus-1/system.d/mcontrolcenter-helper.conf
+    mkdir -p $out/share/dbus-1/system-services
+    echo "[D-BUS Service]" >> $out/share/dbus-1/system-services/mcontrolcenter.helper.service
+    echo "Name=mcontrolcenter.helper" >> $out/share/dbus-1/system-services/mcontrolcenter.helper.service
+    echo "Exec=$out/libexec/mcontrolcenter-helper" >> $out/share/dbus-1/system-services/mcontrolcenter.helper.service
+    echo "User=root" >> $out/share/dbus-1/system-services/mcontrolcenter.helper.service
+  '';
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/mcontrolcenter/derivation.nix:53]:
```suggestion
    runHook postInstall
  '';
```

File: pkgs/os-specific/linux/mcontrolcenter/derivation.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      exec = "mcontrolcenter";
+      icon = "mcontrolcenter";
+      comment = finalAttrs.meta.description;
+      desktopName = "MControlCenter";
+      categories = [ "System" ];
+    })
+  ];
+
+  buildInputs = [
+    pkgs.cmake
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/mcontrolcenter/derivation.nix:19]:
```suggestion
```

File: pkgs/os-specific/linux/mcontrolcenter/derivation.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    qtbase
+  ];
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/mcontrolcenter/derivation.nix:24]:
```suggestion
    cmake
    wrapQtAppsHook
```

File: pkgs/os-specific/linux/mcontrolcenter/derivation.nix [hunk 7]
```diff
@@ -0,0 +1,62 @@
+{ pkgs, lib, stdenv, qtbase, wrapQtAppsHook, qttools, makeDesktopItem, copyDesktopItems, fetchFromGitHub }:
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/mcontrolcenter/derivation.nix:1]:
```suggestion
{ pkgs, lib, stdenv, qtbase, wrapQtAppsHook, qttools, makeDesktopItem, copyDesktopItems, fetchFromGitHub, cmake }:
```

File: pkgs/os-specific/linux/mcontrolcenter/derivation.nix [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+stdenv.mkDerivation (finalAttrs: rec {
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/mcontrolcenter/derivation.nix:3]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/os-specific/linux/mcontrolcenter/default.nix [hunk 9]
```diff
@@ -0,0 +1,2 @@
+{ pkgs ? import <nixpkgs> {} }:
+pkgs.libsForQt5.callPackage ./derivation.nix {}
```
Comment (FliegendeWurst) [inline pkgs/os-specific/linux/mcontrolcenter/default.nix:2]: Usually, this line would be in `pkgs/top-level/all-packages.nix`.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
Sorry for the late review. There are a lot of open PRs and not enough reviewers.

**Comment by Tommimon** [comment]
I would consider closing this pull request. I contacted personally the author and he is not interested in completing it any more, he also encouraged me to take over the work if I have a need for this package. I think is better to move the effort to https://github.com/NixOS/nixpkgs/pull/306216 and try to get this package merged.

**Comment by Tommimon** [comment]
[#306216](https://github.com/NixOS/nixpkgs/pull/306216) has just been merged, mcontrolcenter is now into master, there is no need for this pull request any more, please can we close it?

**Pull request closed without merging**